arch/cpu/nrf/sys/clock-arch.c

```diff
@@ -131,19 +143,20 @@ clock_time(void)
   return (clock_time_t)(ticks & 0xFFFFFFFF);
 }
 /*---------------------------------------------------------------------------*/
-void
+static void
 clock_update(void)
 {
   ticks++;
-  if(etimer_pending()) {
+  if(etimer_pending() &&
+     (etimer_next_expiration_time() - (clock_time_t)ticks - 1) > MAX_TICKS) {
```

Comment:
This looks like some unrelated bug fix, does it belong in a separate PR?

Reply:
Yes, the clock changes will become a separate PR.